openfecwebapp/api_caller.py

```diff
@@ -86,14 +86,19 @@ def load_legal_advisory_opinion(ao_no):
     if not (documents and len(documents)):
         return None
 
-    canonical_document = documents[0]
+    for document in documents:
+        canonical_document = document
+        if (document['category'] == 'Final Opinion'):
```

Comment:
Won't this throw an error if no final opinion is present? I think then canonical_document would be undefined and when you try to index it below it would thrown an error.